### PR TITLE
Option to not substract mean for normalizer and ObservationNormalizer

### PR DESCRIPTION
**Pull request by emailweixu**

### CODE REVIEW

File: alf/utils/normalizers.py [hunk 1]
```diff
@@ -107,19 +113,26 @@ def _summary(name, val):
                         alf.summary.scalar(name + ".min." + suffix, val.min())
                         alf.summary.scalar(name + ".max." + suffix, val.max())
 
-            def _summarize_all(t, m, m2, path):
+            def _summarize_all(t, m2, path, m=None):
                 if path:
```
Comment (hnyu) [inline alf/utils/normalizers.py:117]: I believe that 'path' will always be the last argument?

File: alf/utils/normalizers.py [hunk 2]
```diff
@@ -107,19 +113,26 @@ def _summary(name, val):
...
+                if m is not None:
+                    _summary(path + "mean", m)
+                    _summary(path + "var", m2 - math_ops.square(m))
+                else:
+                    _summary(path + "var", m2)
+
+            if self._mean_averager:
+                alf.nest.py_map_structure_with_path(_summarize_all, tensor,
+                                                    self._mean_averager.get(),
+                                                    self._m2_averager.get())
```
Comment (hnyu) [inline alf/utils/normalizers.py]: mean and m2 order is not correct

**Review by hnyu — CHANGES_REQUESTED** [review]

File: alf/utils/normalizers.py (hunk 2 above)
Comment (le-horizon) [reply alf/utils/normalizers.py]: This is a bit messy, maybe we should have py_map_structure_with_path take a function that has "path" as the first argument:
 _summarize_all(path, t, m2, m=None)

File: alf/utils/normalizers.py [hunk 3]
```diff
@@ -137,18 +150,27 @@ def normalize(self, tensor, clip_value=-1.0):
...
             if clip_value > 0:
                 t = torch.clamp(t, -clip_value, clip_value)
             return t
 
-        return alf.nest.map_structure(_normalize, self._mean_averager.get(),
-                                      self._m2_averager.get(), tensor)
+        if self._mean_averager:
+            return alf.nest.map_structure(_normalize,
+                                          self._mean_averager.get(),
+                                          self._m2_averager.get(), tensor)
```
Comment (le-horizon) [inline alf/utils/normalizers.py]: wrong order here?  Maybe have (t, m2, m=None)?

**Review by le-horizon — COMMENTED** [review]
Almost good.

File: alf/utils/normalizers.py (hunk 2 above)
Comment (emailweixu) [reply alf/utils/normalizers.py]: good point.

File: alf/utils/normalizers.py (hunk 3 above)
Comment (emailweixu) [reply alf/utils/normalizers.py]: fixed

File: alf/utils/normalizers.py [hunk 4]
```diff
@@ -60,6 +61,7 @@ def __init__(self,
             auto_update (bool): If True, automatically update mean and variance
               for each call to `normalize()`. Otherwise, the user needs to call
               `update()`
+            zero_mean (bool): whether to make the normalized value be zero-meane
```
Comment (hnyu) [inline alf/utils/normalizers.py]: zero-mean; should comment that if not zero-mean, then the values are divided by m2 instead of var

File: alf/utils/normalizers.py [hunk 5]
```diff
@@ -107,19 +113,26 @@ def _summary(name, val):
...
-
-            alf.nest.py_map_structure_with_path(_summarize_all, tensor,
-                                                self._mean_averager.get(),
-                                                self._m2_averager.get())
+                         _reduce_along_batch_dims(t, m2, torch.max))
+                if m is not None:
+                    _summary(path + "mean", m)
+                    _summary(path + "var", m2 - math_ops.square(m))
+                else:
+                    _summary(path + "var", m2)
```
Comment (hnyu) [inline alf/utils/normalizers.py]: This quantity is no longer variance. Second moment should be better

File: alf/utils/normalizers.py [hunk 6]
```diff
@@ -169,13 +191,15 @@ def __init__(self,
             auto_update (bool): If True, automatically update mean and variance
               for each call to `normalize()`. Otherwise, the user needs to call
               `update()`
+            zero_mean (bool): whether to make the normalized value be zero-meane
```
Comment (hnyu) [inline alf/utils/normalizers.py]: zero-mean

File: alf/utils/normalizers.py [hunk 7]
```diff
@@ -218,13 +245,15 @@ def __init__(self,
...
```
Comment (hnyu) [inline alf/utils/normalizers.py]: zero-mean

File: alf/utils/normalizers.py [hunk 8]
```diff
@@ -269,6 +301,7 @@ def __init__(self,
...
```
Comment (hnyu) [inline alf/utils/normalizers.py]: zero-mean

**Review by hnyu — CHANGES_REQUESTED** [review]

**Review by le-horizon — APPROVED** [review]
LG, thanks.

**Review by hnyu — APPROVED** [review]
